app/updaters/shopify/product_updater.rb

```diff
@@ -0,0 +1,39 @@
+module Shopify
+  class ProductUpdater
+    def initialize(spree_product_id:, product_klass: Spree::Product,
+                   product_api: ShopifyAPI::Product,
+                   attributor: Shopify::ProductAttributes,
+                   exporter: Shopify::ProductExporter)
+
+      @spree_product = product_klass.find(spree_product_id)
+      @product_api = product_api
+      @exporter = exporter
+      @attributor = attributor
+    end
+
+    def save_product_on_shopify
+      shopify_product = find_shopify_product_for(spree_product)
+
+      if shopify_product.persisted?
+        shopify_product.update_attributes(product_attributes)
```

Comment:
Can this be morphed into a private method and then the if else be set to a simple guard clause?